lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -441,9 +441,23 @@ bool LowerClassesPass::shouldCreateClass(FModuleLike moduleLike) {
   if (moduleLike.isPublic())
     return true;
 
-  return llvm::any_of(moduleLike.getPorts(), [](PortInfo port) {
+  // Create a class for modules with property ports.
+  bool hasClassPorts = llvm::any_of(moduleLike.getPorts(), [](PortInfo port) {
     return isa<PropertyType>(port.type);
   });
+
+  if (hasClassPorts)
+    return true;
+
+  // Create a class for modules that instantiate classes or modules with
+  // property ports.
+  for (auto op :
+       moduleLike.getOperation()->getRegion(0).getOps<FInstanceLike>())
```

Comment:
Note to self: this is safe because: for ops like extmodule or intmodule, which have no body, we still create a region, it's just that the region has no blocks.

Also note to self, this will catch firrtl.objects ops as well, including when a firrtl.object has no ports (because the object op has a single result, which is a property).